lms/djangoapps/support/tests/test_views.py

```diff
@@ -329,18 +329,51 @@ def test_get_manual_enrollment_history(self):
 
     @disable_signal(signals, 'post_save')
     @ddt.data('username', 'email')
-    def test_change_enrollment(self, search_string_type):
+    def test_create_new_enrollment(self, search_string_type):
+        testUser = UserFactory.create(username='newStudent', email='test2@example.com', password='test')
+        assert ManualEnrollmentAudit.get_manual_enrollment_by_email(testUser.email) is None
+        url = reverse(
+            'support:enrollment_list',
+            kwargs={'username_or_email': getattr(testUser, search_string_type)}
+        )
+        response = self.client.post(url, data={
+            'course_id': str(self.course.id),
+            'mode': CourseMode.AUDIT,
+            'reason': 'Financial Assistance'
+        })
+        assert response.status_code == 200
+        assert ManualEnrollmentAudit.get_manual_enrollment_by_email(testUser.email) is not None
+
+    @disable_signal(signals, 'post_save')
+    @ddt.data('username', 'email')
+    def test_create_existing_enrollment(self, search_string_type):
         assert ManualEnrollmentAudit.get_manual_enrollment_by_email(self.student.email) is None
         url = reverse(
             'support:enrollment_list',
             kwargs={'username_or_email': getattr(self.student, search_string_type)}
         )
         response = self.client.post(url, data={
+            'course_id': str(self.course.id),
+            'mode': CourseMode.AUDIT,
+            'reason': 'Financial Assistance'
+        })
+        assert response.status_code == 400
+        assert ManualEnrollmentAudit.get_manual_enrollment_by_email(self.student.email) is None
+
+    @disable_signal(signals, 'post_save')
+    @ddt.data('username', 'email')
+    def test_change_enrollment(self, search_string_type):
+        assert ManualEnrollmentAudit.get_manual_enrollment_by_email(self.student.email) is None
+        url = reverse(
+            'support:enrollment_list',
+            kwargs={'username_or_email': getattr(self.student, search_string_type)}
+        )
+        response = self.client.patch(url, data=json.dumps({
             'course_id': str(self.course.id),
             'old_mode': CourseMode.AUDIT,
             'new_mode': CourseMode.VERIFIED,
             'reason': 'Financial Assistance'
-        })
+        }), content_type='application/json')
```

Comment:
Why is the data being dumped here? why it worked with post before and not patch?

Reply:
Received this error without dumping the data: 
`rest_framework.exceptions.UnsupportedMediaType: Unsupported media type "application/octet-stream" in request.`

This means that the request has been sent as "application/octet-stream" which is Django's test default.

The options were either use `APITestCase` or manually dump the data and setting content_type to `application/json`